### DISCUSSION TITLE
City-level (or even deeper) regions

### DISCUSSION THREAD

**Discussion by SaadBazaz** [medusajs/medusa#2846] [Feature Requests]
I don't know if this belongs here, or on [the Admin Panel](https://github.com/medusajs/admin/issues/840).

A use-case I'm working on really requires a deeper level of region granularity. Is this possible in the latest Medusa? Or can it be worked on?

**Comment by srindom** [comment]
Thanks - we plan to redesign shipping options to have more granular conditions. This will include a way of defining geo zones. 

What is your use case? And how would you ideally want to define your geo zones?

**Reply by lakshmajee** [reply]
I would like to have geo zones, that does
1. Restrict adding product to cart when the user shipping address is not within configured geo zones.
2. Allows admin to be configure geo zones for category or product level. This is needed as few products availability is subjected to location (geo zone).

**Comment by maple-yf** [comment]
这是来自叶枫邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。谢谢！

**Comment by SargeKhan** [comment]
Hey @srindom, is there an update regarding this feature?

**Comment by maple-yf** [comment]
这是来自叶枫邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。谢谢！

**Comment by kb-0912** [comment]
Hey, it's there any update on this?

**Comment by kb-0912** [comment]
Still waiting for this update.

**Comment by maple-yf** [comment]
这是来自叶枫邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。谢谢！

**Comment by maple-yf** [comment]
这是来自叶枫邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。谢谢！

**Comment by webexpr-jdevesvrotte** [comment]
Hello @srindom and the Medusa team,

We're also looking forward to the enhanced shipping options, particularly the ability to include specific regions like Corsica in our geo zones. This feature would greatly improve our shipping accuracy and customer satisfaction by allowing us to manage product availability more effectively based on precise locations.

Thank you for considering this addition!

**Comment by maple-yf** [comment]
这是来自叶枫邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。谢谢！
